executor/adapter.go

```diff
@@ -335,3 +336,36 @@ func IsPointGetWithPKOrUniqueKeyByAutoCommit(ctx context.Context, p plan.Plan) b
 		return false
 	}
 }
+
+// isReadCommittedGet returns true when it's not write operation and in READ COMMITTED isolation level.
+func isReadCommittedGet(ctx context.Context, p plan.Plan) bool {
+	if ctx.GetSessionVars().Systems[variable.TxnIsolation] != ast.ReadCommitted {
+		return false
+	}
+
+	// check plan
+	if proj, ok := p.(*plan.Projection); ok {
```

Comment:
How about check `*plan.Update, *plan.Delete, *plan.Insert` and return false, otherwise, return true?

Reply:
And there is `*plan.Execute` that executes another plan.

Reply:
should ```*plan.anaylze``` be added here?

Reply:
I think we can be more conservative at first. @coocood